src/scripts/main.js

```diff
@@ -1,3 +1,222 @@
 'use strict';
 
-// write your code here
+const field = document.querySelector('tbody');
+const button = document.querySelector('button');
+const scoreField = document.querySelector('.game-score');
+
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+
+let state = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0]];
+
+let newState;
+let isWin = false;
+let score = 0;
+const fieldMatrix = [...field.children].map(row => [...row.children]);
```

Comment:
map method returns new array. It's non-mutating method. You don't need copy
```suggestion
const fieldMatrix = field.children.map(row => [...row.children]);
```

Reply:
Ok, thanks)